src/commonMain/kotlin/fr/acinq/lightning/db/PaymentsDb.kt

```diff
@@ -33,28 +33,28 @@ interface PaymentsDb : IncomingPaymentsDb, OutgoingPaymentsDb {
 
 interface IncomingPaymentsDb {
     /** Add a new expected incoming payment (not yet received). */
-    suspend fun addIncomingPayment(preimage: ByteVector32, origin: IncomingPayment.Origin, createdAt: Long = currentTimestampMillis()): IncomingPayment
+    suspend fun addIncomingPayment(incomingPayment: IncomingPayment)
 
     /** Get information about an incoming payment (paid or not) for the given payment hash, if any. */
-    suspend fun getIncomingPayment(paymentHash: ByteVector32): IncomingPayment?
+    suspend fun getLightningIncomingPayment(paymentHash: ByteVector32): LightningIncomingPayment?
```

Comment:
Note how this method and others below are restricted to `LightningIncomingPayment`, not `IncomingPayment`.